rpcs3/Emu/Cell/Modules/cellGame.cpp

```diff
@@ -448,23 +445,55 @@ error_code cellGameDataCheckCreate2(ppu_thread& ppu, u32 version, vm::cptr<char>
 	strcpy_trunc(cbGet->getParam.dataVersion, psf::get_string(sfo, "APP_VER", ""));
 	strcpy_trunc(cbGet->getParam.titleId, psf::get_string(sfo, "TITLE_ID", ""));
 	strcpy_trunc(cbGet->getParam.title, psf::get_string(sfo, "TITLE", ""));
-	// TODO: write lang titles
-
-	funcStat(ppu, cbResult, cbGet, cbSet);
-
-	if (cbSet->setParam)
+	for (u32 i = 0; i < CELL_HDDGAME_SYSP_LANGUAGE_NUM; i++)
 	{
-		// TODO: write PARAM.SFO from cbSet
-		cellGame.todo("cellGameDataCheckCreate2(): writing PARAM.SFO parameters (addr=0x%x)", cbSet->setParam);
+		strcpy_trunc(cbGet->getParam.titleLang[i], psf::get_string(sfo, fmt::format("TITLE_%02d", i)));
 	}
 
+
+	funcStat(ppu, cbResult, cbGet, cbSet);
+	
+
 	switch ((s32)cbResult->result)
 	{
 	case CELL_GAMEDATA_CBRESULT_OK_CANCEL:
-		// TODO: do not process game data
+		// TODO: do not process game data(directory)
 		cellGame.warning("cellGameDataCheckCreate2(): callback returned CELL_GAMEDATA_CBRESULT_OK_CANCEL");
+		return CELL_OK;
 
-	case CELL_GAMEDATA_CBRESULT_OK:
+		
+	case CELL_GAMEDATA_CBRESULT_OK:	
+		//game confirmed that it wants to create directory
+		if (!fs::is_dir(vfs::get(dir + "/USRDIR")) && !fs::create_path(vfs::get(dir + "/USRDIR")))
+			{
+				cellGame.error("cellGameDataCheckCreate2(): folder creation failed");
+				return CELL_GAME_ERROR_NOSPACE;	//don't know which error. picked one at random
+			}
+		if (cbSet->setParam)
+		{
+			const auto vdir = vfs::get(dir);
+			prm->is_temporary = false;
+
+			//older SDK does not define not settable values, hopefully it doesn't just change some values(overwrite)
+			prm->sfo =
+			{
+				{ "TITLE_ID", psf::string(CELL_GAME_SYSP_TITLEID_SIZE, cbSet->setParam->titleId) },
+				{ "TITLE", psf::string(CELL_GAME_SYSP_TITLE_SIZE, cbSet->setParam->title) },
+				{ "VERSION", psf::string(CELL_GAME_SYSP_VERSION_SIZE, cbSet->setParam->dataVersion) }		
```

Comment:
if i remember correctly it fails to compile when done this way, but using emplace like in 484 we could put anything we want there